### PR TITLE
New features

### PR DESCRIPTION
**Pull request by ghost**
Added a recently opened menu item
Added two dark themes options from CodeMirror
Changed tabs to include filename

### CODE REVIEW

File: app/js/Tab.js [hunk 1]
```diff
@@ -26,7 +26,9 @@ class Tab {
    * @param {Element} container
    */
 
-  constructor(container) {
+  constructor(container, tabName) {
+     console.log(tabName)
```
Comment (wellsjo) [inline app/js/Tab.js]: extra log statement

File: app/js/Tab.js [hunk 2]
```diff
@@ -26,7 +26,9 @@ class Tab {
    * @param {Element} container
    */
 
-  constructor(container) {
+  constructor(container, tabName) {
```
Comment (wellsjo) [inline app/js/Tab.js]: maybe check if the tab name already exists to prevent collisions?

File: app/js/Tab.js [hunk 3]
```diff
@@ -65,10 +67,10 @@ class Tab {
    * @param {Integer} tabId
    */
 
-  static generateTabHeaderTemplate(tabId) {
+  static generateTabHeaderTemplate(tabId, tabName) {
     return `
       <li data-tab="${tabId}">
-        <a href="#tab-${tabId}">&nbsp;</a>
+        <a href="#tab-${tabId}">${tabName || "Untitled-" + tabId }</a>
```
Comment (wellsjo) [inline app/js/Tab.js]: I think `${tabId || tabId + " [no name]"` (as vim does) would look better

File: app/js/system/MainMenu.js [hunk 4]
```diff
@@ -195,6 +209,40 @@ class MainMenu {
     return template
   }
 
+  createRecentSubMenu(updated) {
+    // give properties
+    
+    const recentFiles = this.settings.get('recent')
+    console.log( this , recentFiles)
```
Comment (wellsjo) [inline app/js/system/MainMenu.js]: extra log statement

File: index.html [hunk 5]
```diff
@@ -45,6 +45,9 @@
   <link rel="stylesheet" href="node_modules/codemirror/theme/elegant.css" type="text/css">
   <link rel="stylesheet" href="node_modules/codemirror/theme/neat.css" type="text/css">
   <link rel="stylesheet" href="node_modules/codemirror/theme/neo.css" type="text/css">
+  <link rel="stylesheet" href="node_modules/codemirror/theme/seti.css" type="text/css">
+  <link rel="stylesheet" href="node_modules/codemirror/theme/zenburn.css" type="text/css">
+
```
Comment (wellsjo) [inline index.html]: extra whitespace

**Review by wellsjo — CHANGES_REQUESTED** [review]
looks awesome! I have a few minor comments. 

Maybe add `cntrl+tab` and `cntrl+shift+tab` to switch tabs if you're up to it 😄  - that can always be in a separate issue though

**Comment by nmyvision** [comment]
I'll make the requested changes over the weekend.

File: app/js/Tab.js (hunk 3 above)
Comment (nmyvision) [inline app/js/Tab.js]: why tabId ? The point is to put the filename, if the filename isn't provided then we fall back to the Tab id? should it be ${tabName || tabId + " [no name]"}

Comment (wellsjo) [inline app/js/Tab.js]: I meant `tabName`, my bad

File: app/js/Tab.js (hunk 2 above)
Comment (nmyvision) [inline app/js/Tab.js]: I added title tag to the tab which will show the full filename. I think it is valid to have the same filename in two tabs. For example someone could have two files with the same name in different folders. Changing the display name could be confusing.

File: app/css/ui.css [hunk 6]
```diff
@@ -134,7 +134,7 @@ body {
 }
 
 .ui-tabs-tab {
```
Comment (wellsjo) [inline app/css/ui.css]: looks like we can remove this

File: app/js/system/MainMenu.js [hunk 7]
```diff
@@ -45,12 +44,26 @@ class MainMenu {
             BrowserWindow.getFocusedWindow() || BrowserWindow.getAllWindows()[0],
             { filters: [{ name: 'JSON Files', extensions: ['json'] }, { name: 'All Files', extensions: ['*'] }] },
             (filePaths) => {
+              var filename = filePaths[0]           
+              var r = this.settings.get('recent') || []
+              if (r.indexOf(filename.toLowerCase()) === -1)
```
Comment (wellsjo) [inline app/js/system/MainMenu.js]: bracket should be on same line as if-statement

**Review by wellsjo — CHANGES_REQUESTED** [review]
just a couple minor things, then looks good to go 😄

**Pull request merged**